rs/embedders/src/wasmtime_embedder.rs

```diff
@@ -332,6 +336,42 @@ impl WasmtimeEmbedder {
         self.pre_instantiate(&module)
     }
 
+    /// TODO(EXC-1800): Replace this with `wasmtime::Module::deserialize_open_file`.
+    fn deserialize_from_file(&self, serialized_module: &File) -> HypervisorResult<Module> {
+        let mmap_size = serialized_module.metadata().unwrap().size() as usize;
+        let mmap_ptr = unsafe {
+            mmap(
+                std::ptr::null_mut(),
+                mmap_size,
+                ProtFlags::PROT_READ,
+                MapFlags::MAP_PRIVATE,
+                serialized_module.as_raw_fd(),
+                0,
+            )
+        }
+        .unwrap_or_else(|err| panic!("Module deserialization failed: {:?}", err))
+            as *mut u8;
+        let bytes = unsafe { std::slice::from_raw_parts(mmap_ptr, mmap_size) };
+        unsafe {
```

Comment:
I observe that we have several unsafe blocks closely together. Is there a reason for this? Why not have one larger one?

Reply:
Also in one, the errorhandling (unwrap_or_else) happens outside the unsafe block, whereas here, it is inside (map_err). Any particular reason? (This is not a request for change)

Reply:
Generally I'd say the `unsafe` blocks should be as small as possible/reasonable so that it's clear exactly what's going on that's unsafe. In this case I'd probably say we move the `map_err` outside the block.